independent-projects/arc/tests/src/test/java/io/quarkus/arc/test/observers/async/AsyncObserverExceptionTest.java

```diff
@@ -31,19 +32,28 @@
     public ArcTestContainer container = new ArcTestContainer(StringProducer.class, StringObserver.class);
 
     @Test
-    public void testAsyncObservers() throws InterruptedException, ExecutionException, TimeoutException {
+    public void testAsyncObservers() throws InterruptedException {
         ArcContainer container = Arc.container();
         StringProducer producer = container.instance(StringProducer.class).get();
         StringObserver observer = container.instance(StringObserver.class).get();
 
+        final List<Throwable> suppressed = new ArrayList<>();
         BlockingQueue<Object> synchronizer = new LinkedBlockingQueue<>();
         producer.produceAsync("pong").exceptionally(ex -> {
+            Arrays.stream(ex.getSuppressed()).forEach(t -> suppressed.add(t));
             synchronizer.add(ex);
             return ex.getMessage();
         });
 
         Object exception = synchronizer.poll(10, TimeUnit.SECONDS);
+
+        // assert suppressed exception is always present
+        assertEquals(1, suppressed.size());
+        assertTrue(suppressed.get(0) instanceof RuntimeException);
+
+        // assert actual cause, always a CompletionException
         assertNotNull(exception);
+        assertTrue(exception instanceof CompletionException);
```

Comment:
I believe that the assertion on the next line should verify the cause instead, i.e. `assertTrue(exception.getCause() instanceof RuntimeException);`

Reply:
True, in case of single exception this is correct.
I will actually also add a test for multiple exceptions thanks for the remainder.